src/heitech.zer0mqXt.core/Patterns/PubSub.cs

```diff
@@ -20,21 +20,41 @@ public PubSub(SocketConfiguration configuration)
             _configuration = configuration;
         }
 
+        private PublisherSocket _publisherSocket;
+
+        internal void PrimePublisher()
+        {
+            try
+            {
+                _publisherSocket = new PublisherSocket();
+                _publisherSocket.Bind(_configuration.Address());
+            }
+            catch (System.Exception ex)
+            {
+                _configuration.Logger.Log(new ErrorLogMsg(ex.Message));
+                throw;
+            }
+
+        }
+
         #region Publishing
         public async Task<XtResult<TMessage>> PublishAsync<TMessage>(TMessage message)
             where TMessage : class, new()
         {
-            PublisherSocket publisherSocket = null;
+            if (_publisherSocket == null)
+            {
+                _configuration.Logger.Log(new ErrorLogMsg("publisherSocket was not primed (setup connection via NetMQ.Bind). When creating a Zer0MQ Bus, make sure to build it with UsePublisher."));
```

Comment:
Perhaps we could store this error message in a local, or const local, and then use it in both the logger and the throw expression to avoid duplication and so that we need to make changes to that message only at one place in the future.